### PR TITLE
Backup/restore KUTTL tests

### PR DESCRIPTION
**Pull request by abays**
Also fixes a few minor bugs related to backup/restore

### CODE REVIEW

File: controllers/openstackclient_controller.go [hunk 1]
```diff
@@ -96,7 +96,10 @@ func (r *OpenStackClientReconciler) Reconcile(ctx context.Context, req ctrl.Requ
 	//
 	// initialize condition
 	//
-	cond := &ospdirectorv1beta1.Condition{}
+	cond := instance.Status.Conditions.GetCurrentCondition()
+	if cond == nil {
+		cond = &ospdirectorv1beta1.Condition{}
+	}
```
Comment (stuggi) [inline controllers/openstackclient_controller.go:102]: how about to create a instance.Status.Conditions.InitCondition() which does these tasks? I mean GetCurrentCondition() and if nil returns &ospdirectorv1beta1.Condition{}

File: controllers/openstackclient_controller.go [hunk 2]
```diff
@@ -139,7 +142,7 @@ func (r *OpenStackClientReconciler) Reconcile(ctx context.Context, req ctrl.Requ
 	}(cond)
 
 	// If we determine that a backup is overriding this reconcile, requeue after a longer delay
-	overrideReconcile, err := ospdirectorv1beta1.OpenStackBackupOverridesReconcile(r.Client, instance.Namespace, true)
+	overrideReconcile, err := ospdirectorv1beta1.OpenStackBackupOverridesReconcile(r.Client, instance.Namespace, instance.Status.Conditions.GetCurrentCondition() != nil && instance.Status.Conditions.GetCurrentCondition().Reason == ospdirectorv1beta1.OsClientCondReasonPodProvisioned)
```
Comment (stuggi) [inline controllers/openstackclient_controller.go:145]: with the above `cond := instance.Status.Conditions.GetCurrentCondition()` shouldn't it be ok to just check cond.Reason? it should never be nil here, right?

File: controllers/openstackconfiggenerator.go [hunk 3]
```diff
@@ -121,7 +121,10 @@ func (r *OpenStackConfigGeneratorReconciler) Reconcile(ctx context.Context, req
 	//
 	// initialize condition
 	//
-	cond := &ospdirectorv1beta1.Condition{}
+	cond := instance.Status.Conditions.GetCurrentCondition()
+	if cond == nil {
+		cond = &ospdirectorv1beta1.Condition{}
+	}
```
Comment (stuggi) [inline controllers/openstackconfiggenerator.go:127]: could then use in all those just instance.Status.Conditions.InitCondition()

File: controllers/openstackvmset_controller.go [hunk 4]
```diff
@@ -223,7 +226,7 @@ func (r *OpenStackVMSetReconciler) Reconcile(ctx context.Context, req ctrl.Reque
 	}
 
 	// If we determine that a backup is overriding this reconcile, requeue after a longer delay
-	overrideReconcile, err := ospdirectorv1beta1.OpenStackBackupOverridesReconcile(r.Client, instance.Namespace, instance.Status.ProvisioningStatus.State == ospdirectorv1beta1.VMSetCondTypeProvisioned)
+	overrideReconcile, err := ospdirectorv1beta1.OpenStackBackupOverridesReconcile(r.Client, instance.Namespace, instance.Status.ProvisioningStatus.State == ospdirectorv1beta1.VMSetCondTypeProvisioned || instance.Status.ProvisioningStatus.State == ospdirectorv1beta1.VMSetCondTypeEmpty)
```
Comment (stuggi) [inline controllers/openstackvmset_controller.go:229]: as we discussed a while back to get rid of the dedicated ProvisioningStatus, could use cond here which is a this point currentCondition?

File: controllers/openstackclient_controller.go (hunk 1 above)
Comment (abays) [reply controllers/openstackclient_controller.go:102]: Good idea

File: controllers/openstackclient_controller.go (hunk 2 above)
Comment (abays) [reply controllers/openstackclient_controller.go:145]: Yep, and that will be cleaner to read as well

File: controllers/openstackconfiggenerator.go (hunk 3 above)
Comment (abays) [reply controllers/openstackconfiggenerator.go:127]: Ack

File: controllers/openstackvmset_controller.go (hunk 4 above)
Comment (abays) [reply controllers/openstackvmset_controller.go:229]: That is the proper approach.  Will fix.

File: controllers/openstackclient_controller.go (hunk 1 above)
Comment (abays) [reply controllers/openstackclient_controller.go]: Done

File: controllers/openstackclient_controller.go (hunk 2 above)
Comment (abays) [reply controllers/openstackclient_controller.go]: Done

File: controllers/openstackvmset_controller.go (hunk 4 above)
Comment (abays) [reply controllers/openstackvmset_controller.go]: Done

**Review by stuggi — APPROVED** [review]
/lgtm

**Pull request merged**